internal/providers/powervs/powervs.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2021 Red Hat.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The PowerVS (https://www.ibm.com/products/power-virtual-server) provider fetches configurations from the userdata available in
+// the config-drive. Power VS is based on the ibmcloud vpc gen1 which is similar to Openstack and uses the same APIs
+
+package powervs
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/coreos/ignition/v2/config/v3_3_experimental/types"
+	"github.com/coreos/ignition/v2/internal/distro"
+	"github.com/coreos/ignition/v2/internal/log"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+
+	"github.com/coreos/vcontext/report"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	configDriveUserdataPath = "/openstack/latest/user_data"
+)
+
+func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
```

Comment:
If we're only going to support one config source, we can remove the parallelism and timeout code in this function.

Reply:
See also https://github.com/coreos/ignition/pull/1221.

Reply:
ah..ok..will do...so in that case...should we wait for the config drive indefinitely ? the assumption is that there will always be one if the platform is `powervs` ?

Reply:
Yes, that's right; Ignition waits indefinitely for the config source.  OpenStack was a special case for a while, but that's been fixed now.

Reply:
done.

Reply:
The dispatch function and cancellation are part of the parallelism code.  This can be trimmed down to just the `fetchConfigFromDevice` and `ParseConfig`, and the `Context` can be dropped from the whole file.

Reply:
ah..correct..changed.